src/popper.ts

```diff
@@ -1,37 +1,41 @@
+import Popper from 'popper.js'
 import Selectors from './selectors'
 import { arrayFrom } from './ponyfills'
 import { innerHTML, div } from './utils'
 import { isUCBrowser } from './browser'
+import {
+  PopperElement,
+  Props,
+  PopperChildren,
+  HideAllOptions,
+  BasicPlacement,
+} from './types'
 
 /**
- * Sets the innerHTML of an element while tricking linters & minifiers
- * @param {HTMLElement} element
- * @param {Element|String} html
+ * Sets the innerHTML of an element
  */
-export function setInnerHTML(element, html) {
-  element[innerHTML()] = html instanceof Element ? html[innerHTML()] : html
+export function setInnerHTML(element: Element, html: string | Element) {
+  ;(element as any)[innerHTML()] =
+    html instanceof Element ? (html as any)[innerHTML()] : html
 }
 
 /**
  * Sets the content of a tooltip
- * @param {HTMLDivElement} contentEl
- * @param {Object} props
  */
-export function setContent(contentEl, props) {
+export function setContent(contentEl: PopperChildren['content'], props: Props) {
   if (props.content instanceof Element) {
     setInnerHTML(contentEl, '')
     contentEl.appendChild(props.content)
   } else {
+    // @ts-ignore
     contentEl[props.allowHTML ? 'innerHTML' : 'textContent'] = props.content
```

Comment:
The warning is correct. What if `props.content` is a function?

To avoid unknown index, you can use
```ts
const key: 'innerHTML' | 'textContent' = props.allowHTML ? 'innerHTML' : 'textContent'
contentEl[key] = props.content
```